packages/ts/react-crud/src/autoform.tsx

```diff
@@ -185,7 +186,7 @@ export function AutoForm<M extends AbstractModel>({
       const newItem = await form.submit();
       if (newItem === undefined) {
         // If update returns an empty object, then no update was performed
-        throw new Error('generic error');
+        throw new EndpointError('No update performed');
       } else if (afterSubmit) {
         afterSubmit({ item: newItem });
```

Comment:
This can still result in an error, and would now cause the `Something went wrong, please check all your values` text to show up in the form. That doesn't seem right. I think we could:
- Either wrap `afterSubmit` into a try/catch and swallow the error
- Or do error handling only for `EndpointError`s, and rethrow the error otherwise. Something like:
```ts
if (error instanceof ValidationError) {
  // Handled automatically
  return;
}
if (error instanceof EndpointError) {
  if (onSubmitError) {
    onSubmitError({ error });
  } else {
    setFormError('Something went wrong, please check all your values');
  }
}
// Rethrow
throw error;
```

The second option seems preferrable IMO. It doesn't swallow errors from application code in `afterSubmit`, and all of the error handling in the component becomes explicit about endpoint errors.

Reply:
I agree and committed it. I just changed it so that the error message is used instead of that default.